### PR TITLE
Aida 815

### PR DESCRIPTION
**Pull request by sharknc**
- Created new batch-single Docker container that combines both the main and worker python scripts into a single docker container
- Created entrypoint.py script that will run either main.py or worker.py based on the batch node index that is read in from the environment variable when running on AWS batch
- Updated the init.py script to now be within a Docker container 
- Updated the init.py script the automatically submit batch jobs based on the submission that was extracted

Updated the main.py script to fix the following:
1. Removed the functionality of trying to identify if a Task is type 1a or 1b, they are all now just considered Task 1. 
2. Updated the reports to be a little more concise and clear. 
3. Created a new file validation.report which is included in the resulting tar that contains the validation report summary that is also sent via SNS
4. Cleaned up some of the report generation code 

This should be test on AWS batch. To rest you will need to build the init Docker image and run it via the run.sh script. Because I have not created a CloudFormation script to create the batch-single stack, you will need to run the run.sh script with the following environment variables to test:

BATCH_NUM_NODES='2'
BATCH_JOB_DEFINITION='aida-single-node-nist:10`
BATCH_JOB_QUEUE='aida-validation-cf-queue'

### CODE REVIEW

**Comment by sharknc** [comment]
No, I missed that was committed. I’ll update it and merge with master to resolve the conflicts.

On Jun 7, 2019, at 3:23 PM, Craig Warsaw <notifications@github.com<mailto:notifications@github.com>> wrote:


@craigwarsaw commented on this pull request.

Did you really mean to check in your .worker.py.swp file?
There were some merge conflicts.

I would like to look at this more, but submitting this now

File: docker/init/scripts/init.py [hunk 1]
```diff
@@ -422,4 +523,4 @@ def main():
 		raise ValueError("Exception occurred when validating environment variables") 
 
 
-if __name__ == "__main__": main()
+if __name__ == "__main__": main()
```
Comment (jakevb) [inline docker/init/scripts/init.py]: Add newline at the end of the file.

File: docker/batch-single/scripts/worker.py [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+			messages = sqs_client.receive_message(
+				QueueUrl=queue_url,
+				MaxNumberOfMessages=1,	# only recieve a single message
+				WaitTimeSeconds=20, 	# enable long polling
+				VisibilityTimeout=10    
+			)
+
+			if 'Messages' in messages:
+				msg = messages['Messages'][0]
+				logging.info("Recieved message with receipt handle %s", msg['ReceiptHandle'])
```
Comment (jakevb) [inline docker/batch-single/scripts/worker.py]: Typo: Recieved should be Received.

File: docker/batch-single/scripts/main.py [hunk 3]
```diff
@@ -0,0 +1,734 @@
...
+    if validate_envs(envs):
+
+        main = Main(envs)
+        main.run()
+
+    else:
+        raise ValueError("Exception occurred when validating environment variables") 
+
+
+if __name__ == "__main__": main()
```
Comment (jakevb) [inline docker/batch-single/scripts/main.py]: Add newline at the end of the file.

File: docker/batch-single/scripts/main.py [hunk 4]
```diff
@@ -0,0 +1,734 @@
...
+
+
+    def _enqueue_files(self, queue_url, objects):
+        """Uploads all turtle (ttl) files in source directory to the provided S3 bucket,
+        adds each S3 object path as a message on SQS and creates / updates the source log on S3. 
+        The source log is a serialized list of each S3 object path that will
+        be used for processing validation. After all messages have processed and added to 
+        the queue, a final source file will be uploaded with a suffix of '.queued' and the old
+        source file will be removed from S3.
+        :param str bucket_direcotry: The bucket and prefix directory that contains the unprocessed
```
Comment (jakevb) [inline docker/batch-single/scripts/main.py]: parameter bucket_directory doesn't exist.  Should be replaced with objects and updated to refer to a list of buckets.

File: docker/batch-single/scripts/worker.py [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+    if validate_envs(envs):
+
+        worker = Worker(envs)
+        worker.run()
+
+    else:
+        raise ValueError("Exception occured when validating environment variables") 
+
+
+if __name__ == "__main__": main()
```
Comment (jakevb) [inline docker/batch-single/scripts/worker.py]: Add newline at the end of the file.

File: docker/init/scripts/init.py (hunk 1 above)
Comment (sharknc) [reply docker/init/scripts/init.py]: fixed

**Comment by sharknc** [comment]
> Moving the docker/main and docker/worker folders into a separate folder called batch-multi or something would be nice to reduce possible confusion from the similar main and worker in the batch-single folder.

I moved worker and main into batch-multi directory. If we eventually decide to just stick with the batch-single approach I will mark this directory for deletion in a future JIRA ticket.

File: docker/batch-single/scripts/worker.py (hunk 5 above)
Comment (sharknc) [reply docker/batch-single/scripts/worker.py]: fixed

File: docker/batch-single/scripts/main.py (hunk 4 above)
Comment (sharknc) [reply docker/batch-single/scripts/main.py]: Updated documentation.

File: docker/batch-single/scripts/main.py (hunk 3 above)
Comment (sharknc) [reply docker/batch-single/scripts/main.py]: fixed

File: docker/batch-single/scripts/worker.py (hunk 2 above)
Comment (sharknc) [reply docker/batch-single/scripts/worker.py]: fixed

**Comment by sharknc** [comment]
@jakevb I made one additional update based on my mistake of pushing aws keys to this repo last week. All run.sh.example scripts have been updated to read credentials directly from the aws cli. If you want to take a look you can see these changes in the following files:

- batch-multi/main/run.sh.example
- batch-multi/worker/run.sh.example
- batch-single/run.sh.example
- init/run.sh.example.

**Pull request merged**